Common/src/Pollster/Agent.cs

```diff
@@ -116,27 +121,90 @@ public async Task FinalizeTaskCreation(CancellationToken cancellationToken)
 
     logger_.LogDebug("Finalize child task creation");
 
+    await resultTable_.Create(createdResults_.AsICollection(),
+                              cancellationToken)
+                      .ConfigureAwait(false);
+
+    await TaskLifeCycleHelper.CreateTasks(taskTable_,
+                                          resultTable_,
+                                          sessionData_.SessionId,
+                                          taskData_.TaskId,
+                                          createdTasks_.AsICollection(),
+                                          logger_,
+                                          cancellationToken)
+                             .ConfigureAwait(false);
+
     await submitter_.FinalizeTaskCreation(createdTasks_,
                                           sessionData_,
                                           taskData_.TaskId,
                                           cancellationToken)
                     .ConfigureAwait(false);
 
-    foreach (var (result, (id, size)) in sentResults_)
+    var resultsToComplete = new Dictionary<string, (byte[] id, long size)>();
+
+    foreach (var result in notifiedResults_)
     {
-      await resultTable_.CompleteResult(taskData_.SessionId,
-                                        result,
-                                        size,
-                                        id,
-                                        cancellationToken)
-                        .ConfigureAwait(false);
+      await using var fs = new FileStream(Path.Combine(Folder,
+                                                       result),
+                                          FileMode.OpenOrCreate);
+      using var r       = new BinaryReader(fs);
+      var       channel = Channel.CreateUnbounded<ReadOnlyMemory<byte>>();
```

Comment:
I don't think that's a good idea to have an unbounded channel here. We don't want to materialize the whole file in memory in case the upload to the Object Storage is slower than reading the file from disk (which it most likely is). I don't know what the capacity should be, but I think a handful of chunk is enough.

```suggestion
      var       channel = Channel.CreateBounded<ReadOnlyMemory<byte>>(5);
```